src/main/java/org/candlepin/service/model/Event.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+/**
+ * The Event interface represents a Candlepin event.
+ */
+public interface Event extends ServiceAdapterModel {
+
+    /**
+     * @return the event serialized in a format such as JSON.
```

Comment:
Docblock needs more details here to define what is expected from a proper implementation. Is null something it can return? Is there a way to determine the output format without parsing the output directly? Are there any lifecycle concerns an implementer or user would need to be aware of? etc.

Also, as a grammar thing, the existing line isn't a complete sentence (and annotated lines generally won't be) so it shouldn't end with a period.

Reply:
Yeah, you are right. The docs for these interfaces are unclear. For this method, I thought it might make more sense to return an `EventBody` object that encapsulates the serialized text along with the serialization type. That way we can enforce the null and empty string constraints of the event's body in that class instead of hoping the implementers of the Event interface follow the instructions on the JavaDoc.